src/DependencyInjection/Configuration.php

```diff
@@ -32,6 +32,7 @@ public function getConfigTreeBuilder()
         $rootNode
             ->children()
                 ->scalarNode('root_namespace')->defaultValue('App')->end()
+                ->booleanNode('typed_properties')->defaultFalse()->end()
```

Comment:
I think this should default to `null`, meaning that the user has no "preference". Then, we should inject this into the `PhpCompatUtil` class and use it in `canUseTypedProperties`. Basically, if this option is `null`, then that method will do what it does now: return true if php 7.4 or higher is used. If this option IS passed (either as true or false), then that value would be returned directly from that method.

Then we can continue using the `canUseTypedProperties()` method everywhere to figure out if we should use typed properties.

Reply:
Humm for ``canUseTypedProperties`` I'm not agree, this function check on the composer.lock if the keys ['platform-overrides']['php'] exist but on the most php project this key is not defined, and if the key is not defind the function compare the version of the PHP Interpretor and this is a problem in my opinion :/ If you don't see any problem with this I can still use ``canUseTypedProperties``